support/git.mk

```diff
@@ -0,0 +1,125 @@
+########################################################################
```

Comment:
Why is this file part of this pull request?

Reply:
Probably could be a separate PR but I am reintroducing the git.mk since I am using the mechanism implemented in flutter.mk to update the version sequence number in pubspec.yaml (the +5 in the version). This requires the `git push` to be intercepted in flutter.mk and so using the git.mk is one approach to enabling this. Each `make push` updates the sequence number. Let's see how it works but I was testing it for this branch, thus in this PR.